File: IPL_WEEK_8/B/Main.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int n = in.nextInt();
        int m = in.nextInt();
        boolean[] bulbs = new boolean[m];

        for(int i = 0; i < n; i++) {
            int numBulbs = in.nextInt();
            for(int j = 0; j < numBulbs; j++) {
                int tmp = in.nextInt()-1;
                System.out.println(tmp);
                bulbs[tmp] = true;
            }
        }

        for(int i = 0; i < bulbs.length; i++) {
            if(!bulbs[i]) {
                System.out.println("NO");
                return;
            }
        }
        System.out.println("YES");
        return;
    }
}

File: IPL_WEEK_8/B/main.cpp
#include<iostream>
#include<vector>
using namespace std;

int main() {
	int n,m;
	cin >> n >> m;
	bool b[m];

	for(int i = 0; i < m; i++) {
		b[i] = false;
	}
	for(int i = 0; i < n; i++) {
		int l;
		cin >> l;
		for(int j = 0; j < l; j++) {
			int t; 
			cin >> t;
			b[t-1] = true;
		}
	}

	for(int i = 0; i < m; i++) {
		if(b[i] == false) {
			cout << "NO";
			return 0;
		}
	}

	cout << "YES";
	return 0;
}

